src/main.js

```diff
@@ -1,26 +1,166 @@
+*/
+
+// Global variables - DOM elements
+var view = {
+  home: document.querySelector('.home-view'),
+  saved: document.querySelector('.saved-view'),
+  form: document.querySelector('.form-view')
+}
+
+var book = {
+  image: document.querySelector('.cover-image'),
+  title: document.querySelector('.cover-title'),
+  descriptor1: document.querySelector('.tagline-1'),
+  descriptor2: document.querySelector('.tagline-2')
+}
+
+var button = {
+  home: document.querySelector('.home-button'),
+  random: document.querySelector('.random-cover-button'),
+  save: document.querySelector('.save-cover-button'),
+  viewSaved: document.querySelector('.view-saved-button'),
+  makeNew: document.querySelector('.make-new-button')
+}
+
+var data = {
+  currentCover: 'Easter Egg',
+  savedCovers: [
+    new Cover("http://3.bp.blogspot.com/-iE4p9grvfpQ/VSfZT0vH2UI/AAAAAAAANq8/wwQZssi-V5g/s1600/Do%2BNot%2BForsake%2BMe%2B-%2BImage.jpg", "Sunsets and Sorrows", "sunsets", "sorrows")
+  ]
+}
+
+//Event listeners
+window.addEventListener('load', displayRandomBook);
+button.random.addEventListener('click', displayRandomBook);
+button.save.addEventListener('click', function (){});
+
+button.home.addEventListener('click', function(){
+  hideHomeButton();
+  hideSavedView();
+  hideFormView();
+  showRandomButton();
+  showSaveButton();
+  showSavedCoversButton();
+  showFormButton();
+  showHomeView();
+});
+
+button.viewSaved.addEventListener('click', function (){
+  hideRandomButton();
+  hideSaveButton();
+  hideSavedCoversButton();
+  hideHomeView();
+  hideFormView();
+  showHomeButton();
+  showFormButton();
+  showSavedView();
+});
+
+button.makeNew.addEventListener('click', function (){
+  hideRandomButton();
+  hideSaveButton();
+  hideFormButton();
+  hideHomeView();
+  hideSavedView();
+  showHomeButton();
+  showSavedCoversButton();
+  showFormView();
+});
+
+//Functions (maybe could be two function generateRandomCover, displayCover)
 function displayRandomBook() {
-  bookImage.src = covers[getRandomIndex(covers)];
-  bookTitle.innerText = titles[getRandomIndex(titles)];
-  bookDescriptor1.innerText = descriptors[getRandomIndex(descriptors)];
-  bookDescriptor2.innerText = descriptors[getRandomIndex(descriptors)];
-  currentCover = new Cover(bookImage.src, bookTitle.innerText, bookDescriptor1.innerText, bookDescriptor2.innerText);
+  book.image.src = covers[getRandomIndex(covers)];
+  book.title.innerText = titles[getRandomIndex(titles)];
+  book.descriptor1.innerText = descriptors[getRandomIndex(descriptors)];
+  book.descriptor2.innerText = descriptors[getRandomIndex(descriptors)];
+  currentCover = new Cover(book.image.src, book.title.innerText, book.descriptor1.innerText, book.descriptor2.innerText);
 }
-// We've provided one function to get you started
 function getRandomIndex(array) {
   return Math.floor(Math.random() * array.length);
-}
\ No newline at end of file
+}
+function showHomeView() {
```

Comment:
Just on first pass all of these functions are structured very similarly and while they aren't inherently "duplicates" it leads me to wonder if/how you might be able to leverage a more flexible function to pair down the number of individual ones you need built out here.